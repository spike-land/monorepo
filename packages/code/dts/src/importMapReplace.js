import { oo } from "./importMap";
export function importMapReplace(code, origin) {
  if (code.includes("importMapReplace")) {
    return code;
  }
  const topLevelImportPattern =
    /(import\s*(?:[\w{},*\s]+|[\w{} as,*\s|\$]+|\w+|\$|\$\w+)\s*from\s*)(['"`][^'`"]+['"`])/g;
  const topLevelExportPattern =
    /(export\s*(?:[\w{},*\s]+|[\w{} as,*\s|\$]+|\w+|\$|\$\w+)\s*from\s*)(['"`][^'`"]+['"`])/g;
  const dynamicImportPattern = /(import\()(['"`][^'`"]+['"`])(\))/g;
  const replacer = (match, p1, p2, p3char) => {
    const p3 = String(p3char).replace(/[0-9]/g, "");
    const packageName = p2.slice(1, -1);
    if (packageName.startsWith("data:text")) {
      return p1 + `"${packageName}/index.js"` + p3;
    }
    if (
      packageName.startsWith(`${origin}/live`)
      && !packageName.includes("index.js")
    ) {
      return p1 + `"${packageName}/index.js"` + p3;
    }
    if (packageName.startsWith("/")) {
      return p1 + `"${origin}${packageName}"` + p3;
    }
    if (packageName.startsWith(".") || packageName.startsWith("http")) {
      if (packageName.startsWith("http") && !packageName.startsWith(origin)) {
        const oldUrl = new URL(packageName);
        const newPackageName = new URL(oldUrl.pathname, origin);
        return p1 + `"${newPackageName.toString()}"` + p3;
      }
      return match;
    }
    if (packageName.startsWith("/live")) {
      return p1 + `"${origin}${packageName}/index.js"` + p3;
    }
    if (packageName.includes(".")) {
      return p1 + `"${origin}/${packageName}"` + p3;
    }
    return p1 + `"${origin}/*${packageName}?bundle"` + p3;
  };
  const str = typeof code === "string"
    ? code
    : new TextDecoder().decode(new Uint8Array(code));
  let replaced = str
    .replace(topLevelImportPattern, replacer)
    .replace(topLevelExportPattern, replacer)
    .replace(dynamicImportPattern, replacer);
  Object.keys(oo).forEach((pkg) => {
    replaced = replaced.split(`${origin}/*${pkg}?bundle`).join(origin + oo[pkg]);
  });
  Object.keys(oo).forEach((pkg) => {
    replaced = replaced.split(`"${pkg}"`).join("\"" + origin + oo[pkg] + "\"");
  });
  return `
  /** importMapReplace */
  ` + replaced;
}
