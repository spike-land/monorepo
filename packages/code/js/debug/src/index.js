module.exports = require('./browser.js');