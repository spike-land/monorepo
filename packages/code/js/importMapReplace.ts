// import { skipImportmapReplaceNames } from "./esbuildEsm";
import imap from "./importmap.json";

const importMapImports = imap.imports;
export function importMapReplace(
  codeInp: string,
  origin: string,
  relativeUrl: string,
  importmapRep = true,
) {
  // if (skipImportmapReplaceNames) {
  //   return codeInp;
  // }

  let returnStr = replaceAll(codeInp, `from"`, `from "`);

  const items = Object.keys(
    importMapImports,
  ) as (keyof typeof importMapImports)[];

  items.map((lib: keyof typeof importMapImports) => {
    const uri = (new URL(importMapImports[lib], origin)).toString();
    if (importmapRep) {
      returnStr = replaceAll(returnStr, ` from "${lib}"`, ` from "${uri}"`);
    }
    returnStr = replaceAll(returnStr, ` from "/`, ` from "${origin}/`);
  });

  returnStr.split("/::").join(origin);
  if (!returnStr) return returnStr;
  const url = relativeUrl || origin;
  const baSe = (new URL(".", url)).toString();
  const parent = (new URL("..", url)).toString();
  const gParent = (new URL("../..", url)).toString();
  const ggParent = (new URL("../../..", url)).toString();

  // returnStr = replaceAll(returnStr, `from"`, `from "}`);
  returnStr = replaceAll(
    returnStr,
    `reference path="./`,
    `reference path="${baSe}`,
  );
  returnStr = replaceAll(returnStr, `import"`, `import "`);

  returnStr = replaceAll(returnStr, ` from "../../../`, ` from "${ggParent}`);
  returnStr = replaceAll(
    returnStr,
    `import("../../../`,
    ` import("${ggParent}`,
  );
  returnStr = replaceAll(returnStr, `import("../../`, ` import("${gParent}`);
  returnStr = replaceAll(returnStr, `import("../`, ` import("${parent}`);
  returnStr = replaceAll(returnStr, `import("./`, ` import("${baSe}`);
  returnStr = replaceAll(
    returnStr,
    `import "../../../`,
    ` import "${ggParent}`,
  );
  returnStr = replaceAll(returnStr, `import "../../`, ` import "${gParent}`);
  returnStr = replaceAll(returnStr, `import "../`, ` import "${parent}`);
  returnStr = replaceAll(returnStr, `import "./`, ` import "${baSe}`);
  returnStr = replaceAll(returnStr, ` from "../../`, ` from "${gParent}`);
  returnStr = replaceAll(returnStr, ` from "../`, ` from "${parent}`);
  returnStr = replaceAll(returnStr, ` from "./`, ` from "${baSe}`);

  let oldUrl: URL;
  returnStr = returnStr.split(";").map((x) => x.indexOf("import") !== -1 ? x.trim() : x).map((Y) =>
    Y.split("\n").map((x) => {
      if (x.length === 0 || x.indexOf("import") === -1) return x;
      if (
        x.startsWith("import") && x.indexOf(`"`) !== -1
        && x.indexOf(`https://`) === -1 && x.indexOf(origin) === -1
      ) {
        const slices = x.split(`"`);
        slices[1] = origin + "/" + slices[1] + "?dev&format=es2022";
        return slices.join(`"`);
      }
      if (
        x.indexOf("/node_process.js") !== -1
        || x.indexOf("/node_buffer.js") !== -1
        || x.indexOf("@babel/runtime") !== -1
      ) {
        const slices = x.split(`"`);
        try {
          oldUrl = new URL(slices[1]);
          slices[1] = origin + "/npm:" + oldUrl.pathname;
        } catch {
          console.error("URL ERR", slices[1]);
        }

        return slices.join(`"`);
      }
      if (
        x.indexOf("/npm:/") === -1 && x.startsWith("import")
        && x.indexOf(origin) === -1
      ) {
        const slices = x.split(`"`);
        try {
          oldUrl = new URL(slices[1]);
          if (oldUrl && oldUrl.pathname.indexOf(".") === -1) {
            slices[1] += `/index.js"`;
          }
        } catch {
          console.error("URL ERR", slices[1]);
        }

        return slices.join(`"`);
      }
      return x;
    }).join("\n")
  ).join(";");

  returnStr = returnStr.split("/npm:/npm:").join("/npm:");

  return returnStr;
}
function replaceAll(inp: string, search: string, replace: string) {
  if (!inp) return inp;
  return inp.split(search).join(replace);
}
