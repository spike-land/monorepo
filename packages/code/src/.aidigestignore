files.mds
tw.mjs
lib.ts
tw.html
ipfsWorker.tsx
.aidigestignore 
dist
dts
@
styles
tw.html
tw-chunk-be5bad.js
*.css
*.d.ts
assets
jest.config.js
LICENSE.md
.gitignore
ataWorker.ts
.eslintrc.cjs
rome.json
tailwind.config.js
tsconfig.json
install-components.sh
components.json
.vscode/settings.json
favicon.ico
assets/manifest.webmanifest
__snapshots__
@/**/*
assets/app.css
assets/tw-chunk-be5bad.js
*.spec.ts
__tests__
*.test.ts
emotion*
ata.ts
react*
*.worker.ts
import*
AppTo*
Wrapper*
browserify*
LangCha*
makeSess*
startMonaco.ts
isMobile.tsconfig
*.test.ts
*.test.tsx
*.spec.ts
*.spec.tsx
utils/*.txt