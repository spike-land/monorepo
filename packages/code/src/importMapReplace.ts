import { oo } from "./importMap";
// import type { PackageJson } from 'type-fest'

// const debts: {
//   [pkg: string]: Promise<{ packageName: string; entry: string }>;
// } = {};
export async function importMapReplace(code: string, origin: string): Promise<string> {
  const topLevelImportPattern = /(import\s*(?:[\w{},*\s]+|[\w{} as,*\s]+|\w+)\s*from\s*)(['"`][^'`"]+['"`])/g;
  const topLevelExportPattern = /(export\s*(?:[\w{},*\s]+|[\w{} as,*\s]+|\w+)\s*from\s*)(['"`][^'`"]+['"`])/g;
  const dynamicImportPattern = /(import\()(['"`][^'`"]+['"`])(\))/g;

  const replacer = (match: string, p1: string, p2: string, p3: string) => {
    const packageName = p2.slice(1, -1); // Remove quotes
    if (packageName.startsWith(".") || packageName.startsWith("http") || packageName.startsWith("/")) {
      // Ignore relative and absolute URLs
      return match;
    }
    return p1 + "\"" + `${origin}/*${packageName}?bundle` + "\"" + String(p3).replace(/[0-9]/g, "");
  };

  let str = code;
  if (typeof code !== "string") {
    var uint8array = new TextEncoder().encode(code);
    str = new TextDecoder().decode(uint8array);
  }

  // Apply all replacements
  let replaced = str
    .replace(topLevelImportPattern, replacer)
    .replace(topLevelExportPattern, replacer)
    .replace(dynamicImportPattern, replacer);

  // Apply custom mappings
  Object.keys(oo).forEach((pkg) => {
    replaced = replaced.split(`${origin}/*${pkg}?bundle`).join(origin + oo[pkg as keyof typeof oo]);
  });

  // const debtEntries = await Promise.all(Object.values(debts));
  // for (const {packageName, entry} of debtEntries) {
  //   if (entry && entry.length) {
  //     replaced = replaced.split(`${origin}/*${packageName}?bundle`).join(`${origin}/*${packageName}/${entry}?bundle`);
  //   }
  // }

  return replaced;
}
