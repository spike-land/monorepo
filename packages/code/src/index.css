@import "./app/globals.css";
@import "./assets/app.css";