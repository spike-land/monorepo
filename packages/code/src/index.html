<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta name="theme-color" content="#ffffff" media="(prefers-color-scheme: light)">
    <meta name="theme-color" content="#000000" media="(prefers-color-scheme: dark)">
    <meta name="description" content="Spike.land, an online content Editor with AI support">
    
    <!-- Updated Content Security Policy -->
    <meta http-equiv="Content-Security-Policy" content="
        default-src 'self';
        script-src 'self' https://cdn.jsdelivr.net https://static.cloudflareinsights.com data: https://clerk.spike.land blob: 'unsafe-eval';
        worker-src 'self' blob:;
        style-src 'self' 'unsafe-inline' https://fonts.googleapis.com;
        connect-src 'self' https://cdn.jsdelivr.net blob: https://clerk.spike.land wss://*.peerjs.com;
        font-src 'self' data: blob: https://fonts.gstatic.com;
        img-src 'self' data: blob: https://img.clerk.com https://*.clerk.dev;
    ">

    <link rel="manifest" href="/manifest.json" type="application/manifest+json">
    <link rel="preload" href="/assets/g-chunk-72a597.css" as="style">
    <link rel="preload" href="/assets/app.css" as="style">    

    <link rel="stylesheet" href="/assets/g-chunk-72a597.css">
    <link rel="stylesheet" href="/assets/app.css">

    <!-- injectedCss -->

    <base href="/">
    <link rel="shortcut icon" type="image/png" href="/assets/favicons/chunk-chunk-fe2f7da4f9ccc2.png">
    <title>Instant React Editor</title>
    <script src="/assets/tw-chunk-be5bad.js"></script>
</head>
<body>
    <div id="embed"></div>
    <script type="module" src="/start.mjs"></script>
    
</body>
</html>