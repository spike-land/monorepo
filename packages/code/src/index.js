const defaultExample = `import * as React from "react";\nimport ReactDOM from "react-dom";\n\nconst Counter = (\n    { initial = 0 }\n) => {\n    const [clicks, setClicks] = React.useState(initial);\n\n    return <div>\n        <p>Clicks: <strong>{clicks}</strong></p>\n        <button onClick={() => setClicks(clicks + 1)}>+</button>\n        <button onClick={() => setClicks(clicks - 1)}>-</button>\n    </div>;\n};\n\nReactDOM.render(\n    <Counter initial={0} />,\n    document.getElementById("root")\n);\n`;
const html = (_injection = "")=>`<!DOCTYPE html>\n<html>\n<head>\n  <meta content="text/html;charset=utf-8" http-equiv="Content-Type">\n  <meta name="viewport" content="width=device-width,user-scalable=no,initial-scale=1">\n  <meta content="utf-8" http-equiv="encoding">\n\n \n  <style>\n    html,\n    body,\n    div,\n    span,\n    applet,\n    object,\n    iframe,\n    h1,\n    h2,\n    h3,\n    h4,\n    h5,\n    h6,\n    p,\n    blockquote,\n    pre,\n    a,\n    abbr,\n    acronym,\n    address,\n    big,\n    cite,\n    code,\n    del,\n    dfn,\n    em,\n    img,\n    ins,\n    kbd,\n    q,\n    s,\n    samp,\n    small,\n    strike,\n    strong,\n    sub,\n    sup,\n    tt,\n    var,\n    b,\n    u,\n    i,\n    center,\n    dl,\n    dt,\n    dd,\n    ol,\n    ul,\n    li,\n    fieldset,\n    form,\n    label,\n    legend,\n    table,\n    caption,\n    tbody,\n    tfoot,\n    thead,\n    tr,\n    th,\n    td,\n    article,\n    aside,\n    canvas,\n    details,\n    embed,\n    figure,\n    figcaption,\n    footer,\n    header,\n    hgroup,\n    menu,\n    nav,\n    output,\n    ruby,\n    section,\n    summary,\n    time,\n    mark,\n    audio,\n    video {\n      margin: 0;\n      padding: 0;\n      border: 0;\n      font-size: 100%;\n      font: inherit;\n      vertical-align: baseline\n    }\n\n    article,\n    aside,\n    details,\n    figcaption,\n    figure,\n    footer,\n    header,\n    hgroup,\n    menu,\n    nav,\n    section {\n      display: block\n    }\n\n    body {\n      line-height: 1\n    }\n\n    ol,\n    ul {\n      list-style: none\n    }\n\n    blockquote,\n    q {\n      quotes: none\n    }\n\n    blockquote:before,\n    blockquote:after,\n    q:before,\n    q:after {\n      content: '';\n      content: none\n    }\n\n    table {\n      border-collapse: collapse;\n      border-spacing: 0\n    }\n\n    #container {\n      background-color: #1E1E1E;\n      width: 100vw;\n      height: 100vh;\n      animation-duration: 1s;\n      animation-name: opening;\n    }\n\n    body {\n      overflow: hidden;\n      width: 100%;\n      height: 100vh;\n    }\n\n    @keyframes opening {\n      from {\n        width: 10%;\n        height: 20vh;\n      }\n\n      66% {\n        width: 100%;\n        height: 20vh;\n      }\n\n      to {\n        width: 100%;\n        height: 100vh;\n      }\n    }\n\n\n    #error {\n      display: none;\n      background-color: red;\n      opacity: 0.7;\n    }\n\n    #root {\n      display: none;\n    }\n\n    .draggable {\n      margin: 24px;\n      padding: 32px;\n      position: absolute;\n      touch-action: none;\n      overflow: hidden;\n      z-index: 2;\n      word-wrap: break-word;\n      right: 24px;\n      /* float: right; */\n      /* top: 24px; */\n      /* right: 24px ; */\n      -webkit-transform: translate(0px, 0px);\n      transform: translate(0px, 0px);\n      font-size: 32px;\n      background-color: #ddd;\n      border-radius: 16px;\n      width: fit-content;\n      max-width: 40vw;\n      background: #ddd;\n      box-shadow:\n        0 2.8px 2.2px rgba(0, 0, 0, 0.034),\n        0 6.7px 5.3px rgba(0, 0, 0, 0.048),\n        0 12.5px 10px rgba(0, 0, 0, 0.06),\n        0 22.3px 17.9px rgba(0, 0, 0, 0.072),\n        0 41.8px 33.4px rgba(0, 0, 0, 0.086),\n        0 100px 80px rgba(0, 0, 0, 0.12);\n    }\n\n    .almosthidden {\n      opacity: 0.5;\n    }\n\n    button {\n      font-size: large;\n    }\n  </style>\n\n</head>\n<body>\n  <div id="error" class="draggable"></div>\n  <div id="root" class="draggable resize-drag"></div>\n  <div id="container"></div>\n  <script type="module">\n  ` + _injection + ` \n  </script>\n  <script type="module">\n    import {startMonaco} from "https://unpkg.com/@zedvision/smart-monaco-editor@6.2.3/lib/editor.js"\n    import {run} from "https://unpkg.com/@zedvision/code@6.1.2/code.zed-vision.min.js"\n    run(startMonaco);\n  </script>\n</body>\n</html>`
;
const corsHeaders = {
    "Access-Control-Allow-Origin": "*",
    "Access-Control-Allow-Methods": "GET,HEAD,POST,OPTIONS",
    "Access-Control-Max-Age": "86400"
};
const getInject = (componentToStart, hash)=>{
    if (componentToStart.length < 5) return null;
    const startComponent = "`" + componentToStart + "`";
    return `localStorage.setItem("codeBoXHash", "${hash}"); \nlocalStorage.setItem("${hash}", ${startComponent});`;
};
async function handleRequest(request) {
    if (request.method === "GET") {
        const { hash  } = new URL(request.url);
        let inject = null;
        if (hash !== null && hash.length > 5) {
            const json = await SHATEST.get(hash.substr(1));
            if (json !== null) {
                inject = getInject(JSON.parse(json).code, hash.substr(1));
            }
        }
        if (inject === null) inject = getInject(defaultExample, "startHash");
        return new Response(html(inject), {
            headers: {
                "content-type": "text/html"
            }
        });
    } else if (request.method === "POST") {
        const { headers  } = request;
        const data = await request.json();
        const myBuffer = new TextEncoder().encode(JSON.stringify(data));
        const myDigest = await crypto.subtle.digest({
            name: "SHA-256"
        }, myBuffer);
        const hashArray = Array.from(new Uint8Array(myDigest));
        const hash = hashArray.map((b)=>("00" + b.toString(16)).slice(-2)
        ).join("");
        const smallerKey = hash.substring(0, 7);
        SHATEST.put(smallerKey, myBuffer);
        const resp = new Response(`{"hash":"${smallerKey}"}`);
        resp.headers.append("Access-Control-Allow-Origin", "*");
        resp.headers.append("Access-Control-Allow-Methods", "GET,HEAD,POST,OPTIONS");
        resp.headers.append("Access-Control-Max-Age", "86400");
        return resp;
    }
    return handleOptions(request);
}
function handleOptions(request) {
    let headers;
    let respHeaders;
    return new Response(request.body, {
        headers: {
            ...request.headers,
            ...corsHeaders,
            "Access-Control-Allow-Headers": request.headers.get("Access-Control-Request-Headers")
        }
    });
}
addEventListener("fetch", (event)=>{
    event.respondWith(handleRequest(event.request));
});
