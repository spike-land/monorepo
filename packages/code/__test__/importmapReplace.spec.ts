import { importMapReplace } from ".././src/importMapReplace"; // replace with your actual module

describe("importMapReplace", () => {
  const origin = "http://localhost:3000";

  globalThis.fetch = jest.fn().mockImplementation(() =>
    Promise.resolve({
      ok: true,
      json: () =>
        Promise.resolve({
          "name": "some-module",
          "version": "1.0.0",
          "main": "index.js",
          "browser": "browser.js",
          "module": "module.js",
          "types": "index.d.ts",
          "files": [
            "index.js",
            "browser.js",
            "module.js",
            "index.d.ts",
          ],
        }),
    })
  );

  it("should replace top-level imports", async () => {
    const code = "import React from \"react\";";
    const result = await importMapReplace(code, origin);
    expect(result).toMatchSnapshot();
  });

  it("should replace top-level exports", async () => {
    const code = "export { default } from \"react\";";
    const result = await importMapReplace(code, origin);
    expect(result).toMatchSnapshot();
  });

  it("should replace dynamic imports", async () => {
    const code = "const React = import(\"react\");";
    const result = await importMapReplace(code, origin);
    expect(result).toMatchSnapshot();
  });

  it("should ignore relative URLs", async () => {
    const code = "import MyComponent from \"./MyComponent\";";
    const result = await importMapReplace(code, origin);
    expect(result).toMatchSnapshot();
  });

  it("should ignore absolute URLs", async () => {
    const code = "import MyComponent from \"http://example.com/MyComponent.js\";";
    const result = await importMapReplace(code, origin);
    expect(result).toMatchSnapshot();
  });

  it("should apply custom mappings", async () => {
    const code = "import CustomPackage from \"custom-package\";";
    const result = await importMapReplace(code, origin);
    expect(result).toMatchSnapshot();
  });
  it("should replace dynamic imports", async () => {
    const code = "const module = import(\"some-module\");";
    const result = await importMapReplace(code, origin);
    expect(result).toMatchSnapshot();
  });

  it("should replace dynamic imports with await", async () => {
    const code = "const module = await import(\"some-module\");";
    const result = await importMapReplace(code, origin);
    expect(result).toMatchSnapshot();
  });

  it("should replace dynamic imports inside a function", async () => {
    const code = `
      async function loadModule() {
        const module = await import("some-module");
        // do something with module
      }
    `;

    const result = await importMapReplace(code, origin);
    expect(result).toMatchSnapshot();
  });
});
