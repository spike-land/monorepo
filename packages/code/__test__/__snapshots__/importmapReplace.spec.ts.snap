// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`importMapReplace should apply custom mappings 1`] = `"import CustomPackage from "http://localhost:3000/*custom-package?bundle";"`;

exports[`importMapReplace should ignore absolute URLs 1`] = `"import MyComponent from "http://example.com/MyComponent.js";"`;

exports[`importMapReplace should ignore relative URLs 1`] = `"import MyComponent from "./MyComponent";"`;

exports[`importMapReplace should replace coder44  1`] = `
"
    import { css } from "http://localhost:3000/emotion.mjs";
    import { useEffect, useState, useRef } from "http://localhost:3000/reactMod.mjs";
    import type { FC } from "http://localhost:3000/reactMod.mjs";
    import QRTerminal from "http://localhost:3000/*qrcode-terminal?bundle";
    "
`;

exports[`importMapReplace should replace dynamic imports 1`] = `"const React = import("http://localhost:3000/reactMod.mjs");"`;

exports[`importMapReplace should replace dynamic imports 2`] = `"const module = import("http://localhost:3000/*some-module?bundle");"`;

exports[`importMapReplace should replace dynamic imports inside a function 1`] = `
"
      async function loadModule() {
        const module = await import("http://localhost:3000/*some-module?bundle");
        // do something with module
      }
    "
`;

exports[`importMapReplace should replace dynamic imports with await 1`] = `"const module = await import("http://localhost:3000/*some-module?bundle");"`;

exports[`importMapReplace should replace top-level exports 1`] = `"export { default } from "http://localhost:3000/reactMod.mjs";"`;

exports[`importMapReplace should replace top-level imports 1`] = `"import React from "http://localhost:3000/reactMod.mjs";"`;
